go/vt/vtgate/planbuilder/testdata/filter_cases.json

```diff
@@ -6760,5 +6760,73 @@
         "user.user"
       ]
     }
+  },
+  {
+    "comment": "push filter under aggregation",
+    "query": "select count(*) from user left join user_extra on user.id = user_extra.bar where IFNULL(user_extra.collections_status, 'NOTSET') != 'collections_lock'",
+    "plan": {
+      "QueryType": "SELECT",
+      "Original": "select count(*) from user left join user_extra on user.id = user_extra.bar where IFNULL(user_extra.collections_status, 'NOTSET') != 'collections_lock'",
+      "Instructions": {
+        "OperatorType": "Aggregate",
+        "Variant": "Scalar",
+        "Aggregates": "sum_count_star(0) AS count(*)",
+        "ResultColumns": 1,
+        "Inputs": [
+          {
+            "OperatorType": "Projection",
+            "Expressions": [
+              "[COLUMN 0] * COALESCE([COLUMN 1], INT64(1)) as count(*)",
+              "[COLUMN 2] as collections_status"
```

Comment:
the `collections_status` column doesn't need to be returned here. it's a left over from before the filter was pushed under the projection. would be nice to clean up, but not strictly necessary.